lib/client/weblogin.go

```diff
@@ -350,33 +350,40 @@ func SSHAgentSSOLogin(login SSHLogin) (*auth.SSHLoginResponse, error) {
 
 	// If a command was found to launch the browser, create and start it.
 	var execCmd *exec.Cmd
-	switch runtime.GOOS {
-	// macOS.
-	case teleport.DarwinOS:
-		path, err := exec.LookPath(teleport.OpenBrowserDarwin)
-		if err == nil {
-			execCmd = exec.Command(path, clickableURL)
-		}
-	// Windows.
-	case teleport.WindowsOS:
-		path, err := exec.LookPath(teleport.OpenBrowserWindows)
-		if err == nil {
-			execCmd = exec.Command(path, "url.dll,FileProtocolHandler", clickableURL)
-		}
-	// Linux or any other operating system.
-	default:
-		path, err := exec.LookPath(teleport.OpenBrowserLinux)
-		if err == nil {
-			execCmd = exec.Command(path, clickableURL)
+	if browser != "none" {
+		switch runtime.GOOS {
+		// macOS.
+		case teleport.DarwinOS:
+			path, err := exec.LookPath(teleport.OpenBrowserDarwin)
+			if err == nil {
+				execCmd = exec.Command(path, clickableURL)
+			}
+		// Windows.
+		case teleport.WindowsOS:
+			path, err := exec.LookPath(teleport.OpenBrowserWindows)
+			if err == nil {
+				execCmd = exec.Command(path, "url.dll,FileProtocolHandler", clickableURL)
+			}
+		// Linux or any other operating system.
+		default:
+			path, err := exec.LookPath(teleport.OpenBrowserLinux)
+			if err == nil {
+				execCmd = exec.Command(path, clickableURL)
+			}
 		}
 	}
 	if execCmd != nil {
 		execCmd.Start()
 	}
 
-	// Print to screen in-case the command that launches the browser did not run.
-	fmt.Printf("If browser window does not open automatically, open it by ")
-	fmt.Printf("clicking on the link:\n %v\n", clickableURL)
+	// Print the URL to the screen, in case the command that launches the browser did not run.
+	// If Browser is set to the special string teleport.BrowserNone, no browser will be opened.
+	if browser == "none" {
```

Comment:
nit: `if browser == teleport.BrowserNone {`

Reply:
Sorry maybe I'm missing something. BrowserNone changed to just Browser to incorporate the string. It was BrowserNone when it was a bool. I just noticed though, I didn't change the comment above the if statement to reflect that.

Reply:
Nope I'm dumb. Nevermind, I didn't see the const value.